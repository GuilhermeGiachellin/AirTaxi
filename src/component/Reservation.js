/* eslint-disable no-unused-vars */
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import {
  createReservations, fetchReservations, selectAllReservations, selectReservationById,
} from '../redux/slices/reservationsSlice';

const Reservation = (props) => {
  const dispatch = useDispatch();
  const { id } = useParams();
  const entities = useSelector((state) => selectAllReservations(state));
  useEffect(() => {
    dispatch(fetchReservations(id));
  }, [dispatch]);

  const handleSubmit = async () => {
    dispatch(createReservations(id));
  };

  return (
    <div>
      <p>RESERVAS</p>
      <p>
        Date:
        {/* {entities.reservation_date} */}
        <button type="button" onClick={() => handleSubmit()}>CREATE RESERVE</button>
      </p>
    </div>
  );
};

export default Reservation;
