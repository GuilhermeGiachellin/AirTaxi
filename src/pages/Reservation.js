import React from 'react';
import ReservationManager from '../component/managers/ReservationManager';
import Layout from '../component/wrapper/Layout';

const Reservation = () => (
  <Layout preAnimation="fadeIn">
    <div style={{ width: '100vw' }}>
      <ReservationManager />
    </div>
  </Layout>
);

export default Reservation;
