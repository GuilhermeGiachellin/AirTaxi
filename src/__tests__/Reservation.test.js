import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import '@testing-library/jest-dom/extend-expect';
import Reservation from '../pages/Reservation';

describe('Test Reservation', () => {
  const viewRender = () => {
    const component = render(
      <BrowserRouter>
        <Provider store={store}>
          <Reservation />
        </Provider>
        ,
      </BrowserRouter>,
    );
    return component;
  };

  let component;
  beforeEach(() => {
    component = viewRender();
  });

  test('Render reservation title', async () => {
    const { getByText } = component;
    expect(getByText('BOOK YOUR PLANE')).toBeInTheDocument();
  });

  test('Render reservation button 1', async () => {
    const { getByText } = component;
    expect(getByText('Reserved dates')).toBeInTheDocument();
  });

  test('Render reservation button 2', async () => {
    const { getByText } = component;
    expect(getByText('Book now')).toBeInTheDocument();
  });
});

describe('Snapshoot of the Reservation page', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Reservation />
        </Provider>
        ,
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
