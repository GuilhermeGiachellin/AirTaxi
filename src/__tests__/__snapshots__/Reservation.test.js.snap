// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshoot of the Reservation page renders correctly 1`] = `
Array [
  <main
    style={
      Object {
        "opacity": 0,
        "transform": "none",
      }
    }
  >
    <div
      style={
        Object {
          "width": "100vw",
        }
      }
    >
      <div
        className="container"
      >
        <div
          className="filter"
        >
          <div
            className="content_cnt"
          >
            <button
              className="icon"
              onClick={[Function]}
              type="button"
            >
              <svg
                fill="currentColor"
                height={30}
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                version="1.1"
                viewBox="0 0 16 16"
                width={30}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0.5 8l7.5 7.5v-4.5h8v-6h-8v-4.5z"
                />
              </svg>
            </button>
            <h2
              className="title"
            >
              BOOK YOUR PLANE
            </h2>
            <hr
              className="line"
            />
            <p
              className="description"
            >
              Choose a date to book your plane that you selected. Keep in mind that the plane with all included services are disponible for for 24 hours. You can meet the crew in your local airport ready for the service.
            </p>
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "gap": "2rem",
                  "height": "400",
                }
              }
            />
            <button
              className="button"
              onClick={[Function]}
              type="button"
            >
              Reserved dates
            </button>
            <button
              className="button"
              onClick={[Function]}
              type="button"
            >
              Book now
            </button>
          </div>
        </div>
      </div>
    </div>
  </main>,
  ",",
]
`;
